std/algorithm/iteration.d

```diff
@@ -380,7 +392,7 @@ private struct _Cache(R, bool bidir)
         if (!source.empty)
             caches[0] = source.front;
         else
-            caches = CacheTypes.init;
+            caches[0] = UE.init;
```

Comment:
When running the unittests, the compiler complained here about a missing frame pointer. I don't know, how this is related to the original issue, but I think, it's a bug here anyway, because setting `caches` to `CacheTypes.init` overwrites the back element in a bidirectional range.

Reply:
I was wrong on this. Of course if the range is empty, front and back should both be set to init.

But meanwhile I found out, that `CacheTypes` is `MapResult!(__lambda2, int[])` for a cached map and `CacheTypes.init` is `MapResult(null, null)`, which doesn't look like what want's to have here. I replaced this by explizit initializers now.